src/vision/status_quo/alan_needs_async_in_traits.md

```diff
@@ -0,0 +1,101 @@
+
+## The story
+
+Alan is working on a project with Barbara which has already gotten off to a [somewhat rocky start](https://github.com/rust-lang/wg-async-foundations/pull/111). He is working on abstracting away the HTTP implementation the library uses so that users can provide their own. He wants the user to implement an async trait called `HttpClient` which has one method `perform(request: Request) -> Response`. Alan tries to create the async trait:
+
+```rust,ignore
+trait HttpClient {
+    async fn perform(request: Request) -> Response;
+}
+```
+
+When Alan tries to compile this, he gets an error:
+
+```ignore
+ --> src/lib.rs:2:5
+  |
+2 |     async fn perform(request: Request) -> Response;
+  |     -----^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  |     |
+  |     `async` because of this
+  |
+  = note: `async` trait functions are not currently supported
+  = note: consider using the `async-trait` crate: https://crates.io/crates/async-trait
+```
+
+Alan, using Rust for a little while now, has learned to follow compiler error messages and adds `async-trait` to the dependencies list. Alan follows the README of `async-trait` and gets the code to compile. Alan is somewhat dismayed that the compile times have increased considerably after adding this new dependency.
+
+**TODO::** Maybe add representive pre and post trait compile times here.
+
+After Alan finishes adding the new trait, he shows his work off to Barbara and mentions he's happy with the work but is a little sad that compile times have worsened. Barbara, an experienced Rust developer, knows that using `async-trait` comes with some issues and in this case is especially worried about tying their public API to a third-party dependency. She decides to not worry Alan with this right now. Alan and Barbara are pretty happy with the results and go on to publish their crate which gets lots of users.
+
+Later on, a potential user of the library wants to use their library in a `no_std` context where they will be providing a custom HTTP stack. Alan and Barbara have done a pretty good job of limiting the use of standard library features and think it might be possible to support this use case. However, they quickly run into a show stopper: `async-trait` boxes all of the futures returned from a async trait function. They report this to Alan through an issue.
+
+Alan, feeling (over-) confident in his Rust skills, decides to try to see if he can implement async traits without using `async-trait`. 
+
+```rust 
+struct Request(String);
+
+trait HttpClient {
+   type Response: Future<Output = ()>;
+
+   fn perform(request: Request) -> Self::Response; 
+}
+```
+
+Alan seems to have something working, but when he goes to update the examples of how to implement this trait in his crate's documentation, he realizes that he either needs to:
+
+* use trait object:
+
+  ```rust,ignore
```

Comment:
nit: I don't think ignore is necessary here since none of the code samples are run.

Reply:
Ah, I saw `ignore` being used in other parts, so I added to be sure. :)

Reply:
It used to be require, but we stopped running `mdbook test`